NDTensors/src/lib/UnallocatedArrays/src/abstractfill/abstractfill.jl

```diff
@@ -0,0 +1,38 @@
+## Here are functions specifically defined for UnallocatedArrays
+## not implemented by FillArrays
+## TODO determine min number of functions needed to be forwarded
+alloctype(A::AbstractFill) = alloctype(typeof(A))
+function alloctype(Atype::Type{<:AbstractFill})
+  return get_parameter(Atype, Position{4}())
+end
+
+allocate(A::AbstractFill) = alloctype(A)(parent(A))
+
+set_eltype(T::Type{<:AbstractFill}, elt::Type) = set_parameters(T, Position{1}(), elt)
+set_ndims(T::Type{<:AbstractFill}, n) = set_parameters(T, Position{2}(), n)
+set_axes(T::Type{<:AbstractFill}, ax::Type) = set_parameters(T, Position{3}(), ax)
+
+## With these functions defined I can print UnallocatedArrays
+## compute things like sum and norm, compute the size and length
+@inline Base.axes(A::AbstractFill) = axes(parent(A))
```

Comment:
@mtfishman These functions work fine and extend to `UnallocatedFill` and `UnallocatedZeros` because `parent(a::Fill) = a` which is defined by `AbstractArrays`. But perhaps this is not correct and I should define these functions for each of the Unallocated types?

Reply:
This definition is not a good idea because it is [type piracy](https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-type-piracy).

Reply:
Okay I moved these to each respective file and defined the function on the UnallocatedX type

Reply:
Thanks!